### ISSUE TITLE
Invisible `=` sign

### ISSUE THREAD

**Issue by hauleth** [tonsky/FiraCode#162]
I am using `Terminal GNOME 3.16.2` and when I am trying to use Fira Code 1.102 `=` sign (`0x3d`) is invisible.

![zaznaczenie_457x56_018](https://cloud.githubusercontent.com/assets/291639/14121410/ab33f592-f5f7-11e5-98e7-68c121c66607.png)

**Comment by rahilwazir** [comment]
ping @tonsky

**Comment by tonsky** [comment]
Yeah, guys, I don’t have Gnome terminal to test. Have you considered a possibility there’s a bug in terminal itself? Because Fira Code works really well everywhere else

**Comment by rahilwazir** [comment]
@tonsky I have tested with several other fonts like Hack, Ubunt Mono, Source Code Pro, Andale, DejaVu, Droid Sans Mono, they all working fine.

**Comment by tonsky** [comment]
Any specific sequence where `=` disappears?

**Comment by rahilwazir** [comment]
@tonsky No, just type `=` anywhere in any sequence, it's just displayed as invisible.

**Comment by MadcapJake** [comment]
Gnome Terminal doesn't support ligatures so you should just use Fira Mono with Gnome Terminal.

**Comment by hauleth** [comment]
I have been using FiraCode as it also provides other smaller or bigger changes to the font other than ligatures (ex. Powerline symbols)

--
Łukasz Jan Niemier

Dnia 18 kwi 2016 o godz. 18:10 Jake Russo <notifications@github.com> napisał(a):

**Comment by KaelWD** [comment]
Broken for me in phpstorm as well. 

This is the sequence 
```
= == ===
- -- ---
!= !==
```
![image](https://cloud.githubusercontent.com/assets/16421948/14731301/74f7b3a8-0892-11e6-9501-6a40a988a0af.png)

**Comment by MadcapJake** [comment]
@KaelWD please open a separate issue, PHPStorm is a completely different implementation of ligatures to Gnome Terminal's (also your issue is different just in general).

**Comment by rahilwazir** [comment]
@KaelWD I'm confused?! Whats the issue?

**Comment by KaelWD** [comment]
@MadcapJake Sorry, will do. It just seemed related because it's a similar problem (part or all of the `=` character not rendering). 

@rahilwazir At size 14 (img 1), `=` and `==` display as a short and long dash respectively. At size 12 the spacing between the lines is off, most visible in `!==`.

**Comment by tonsky** [comment]
@KaelWD report to https://youtrack.jetbrains.com/issue/IDEA-127539

@hauleth @rahilwazir is it broken in Gnome terminal only or in all other linux apps as well. What font rendering setting are you using system-wide?

**Comment by rahilwazir** [comment]
@tonsky Only in Gnome Terminal.

Hinting: Maximum/Full
Antialiasing: RGBA

**Comment by ptkato** [comment]
It's also broken in Terminator, the '=' sign is gone and all other ligatures are messed up.

**Comment by silesky** [comment]
=  is still invisible in default Settings on Gnome Terminal. FYI, Gnome Terminal is most popular linux terminal, the one that 75% of people running Linux use.

**Comment by topaxi** [comment]
This is an issue with libvte (the terminal library gnome-terminal, terminator and other terms use).

See here: https://bugzilla.gnome.org/show_bug.cgi?id=584160

**Comment by meskarune** [comment]
Also happens in termite terminal, which also uses libvte. So I just wanted to confirm.

**Comment by concatime** [comment]
Any news? It seems that we're kinda stuck.
(Also had the issue with the `=` inside [Termite](//github.com/thestinger/termite/issues/315))

**Comment by tonsky** [comment]
was this reported to libvte? maybe you should ask them to fix it?

**Comment by klummy** [comment]
I think this is likely related to #247 since Gnome doesn't support ligatures

**Comment by Albert221** [comment]
I've also experienced this issue. I think that using Fira Mono as terminal font is a good way to avoid this problem, but this should be definitely fixed

**Comment by DeemoHarlos** [comment]
![2018-02-14 22-27-53](https://user-images.githubusercontent.com/20572126/36209306-610c76f4-11d6-11e8-979a-46e124890c89.png)
It works fine on my laptop. I do encounter the problem on my previous laptop, and I don't know why it did not work there and why it works here... maybe because of some updated packages, I guess.
Old Laptop: ASUS X550VB with Ubuntu 16.04 LTS
New Laptop: ASUS UX410U with Ubuntu 16.04 LTS

**Comment by kdrag0n** [comment]
Having this issue in every libvte terminal I've tried as well. Fira Mono works fine. Not only is `=` broken, `</a>` is broken and displays ` /b` with the `<=>` overlapping it. Examples:
![image](https://user-images.githubusercontent.com/7930239/36405121-c15c92de-15a3-11e8-9c8e-720e11a588f3.png) `</b>`

![image](https://user-images.githubusercontent.com/7930239/36405127-d237b610-15a3-11e8-917a-83446850441e.png) `<a href="test"> <a></b>`

**Comment by mrworkman** [comment]
I'm having the same issue in gnome-terminal 3.18. It does seem to be fixed in 3.28, but that's not really an option on older distros.

As indicated above, FiraMono works fine, but it uses a lot more vertical space than FiraCode does.

**Comment by cben** [comment]
FWIW, I saw weird char replacement issues in GNOME Terminal 3.28.2 using VTE 0.52.2 (on fod, in only some Fira Code variants:
- only in Fira Code Retina, `|` U+007C VERTICAL LINE displayed exactly as `^`
  ![captura de pantalla de 2018-11-05 23-14-07](https://user-images.githubusercontent.com/273688/48063993-45c7c500-e1cf-11e8-927d-d2f5f134e7dd.png)

- only in Fira Code Bold Italic, `|` U+007C VERTICAL LINE displayed as `&`, and some other chars `+^` displayed as `⌡`...
  ![captura de pantalla de 2018-11-05 23-14-51](https://user-images.githubusercontent.com/273688/48063918-1749ea00-e1cf-11e8-844d-b19e4c0c0ab5.png)

But after playing with antialising settings back and forth, things changed (eg. now retina is good but bold and italic bold show `|` as `&`), so I guess these are not specifically these fonts and chars, but some kind of font cache issues... (?)

**Comment by vinaykumar2491** [comment]
Same issue even now on some ubuntu installations (on both Guake, Gnome Terminal). 
`Fira Mono` works fine though.
